coalahtml/_coalahtml/app/controllers/files.js

```diff
@@ -33,7 +33,27 @@ angular.module('coalaHtmlApp')
     };
     $scope.fileName = $routeParams.fileName || $rootScope.ROOTHOME;
     parseCoalaProject($scope.fileName);
-    $scope.basename = function(path_name){
-      return path_name.split('/').slice(-1)[0];
+
+    var getValidPath = function(path_name) {
+      return path_name || $rootScope.ROOTHOME;
+    };
+    $scope.basename = function(path_name) {
+      return getValidPath(path_name).split('/').slice(-1)[0];
+    };
+    var _slice = function(path_name, start, stop) {
+      return getValidPath(path_name).split('/').slice(start, stop);
+    };
+    var len = function(path_name) {
+      return getValidPath(path_name).split('/').length;
+    };
```

Comment:
Um actually I think I am going to use those. I am implementing more changes. Keeping it for now?